test/integration/src/test/java/io/pravega/test/integration/KeyValueTableTest.java

```diff
@@ -192,6 +261,1427 @@ public void testCreateDeleteKeyValueTable() {
+
+        List<Integer> keyList2 = new ArrayList<>();
+        for (int i = 0; i < keyArray.length; i++) {
+            keyList2.add(keyArray[i]);
+        }
+        log.info("Key list: {}", keyList2);
+        List<TableEntry<Integer, String>> getEntryList3 = newKeyValueTable2.getAll(keyFamily, keyList2).join();
+        assertNotNull("Get Entry List 3 should not be empty", getEntryList3);
+        assertEquals("Get Entry List size 3 should be same as get entry list size",
+                getEntryList3.size(), entryList.size());
+        log.info("Get Entry List: {}", getEntryList3);
+        for (int i = 0; i < keyArray.length; i++) {
+            assertEquals("Corresponding key for getEntryList3 should be as inserted",
+                    keyList2.get(i), getEntryList3.get(i).getKey().getKey());
+            assertEquals("Corresponding key for entryList should be as inserted",
+                    entryList.get(i).getKey().getKey(), getEntryList3.get(i).getKey().getKey());
+        }
+
+        // Comparing each & every entry values of both entrylist and getEntryList.
+        for (int i = 0; i < keyArray.length; i++) {
+            assertEquals("All string values of both entryList and getEntryList3 should be equal",
+                    entryList.get(i).getValue(), getEntryList3.get(i).getValue());
+        }
+        log.info("Successfully completed testing recreation of KVTable with same name");
+    }
+
+    /**
+     * Test concurrent requests to create KVTable with same name +
+     * but different config values
+     */
+    @Test
+    public void testConcurrentCreateTable() throws ExecutionException, InterruptedException {
+        String newScope = "TestScope-" + randomAlphanumeric(5);
+        KeyValueTableInfo newKVT = newKeyValueTableName(newScope);
+        AtomicInteger successCount = new AtomicInteger();
+        AtomicInteger tableExistsCount = new AtomicInteger();
+
+        assertTrue("The requested scope must be created", this.controller.createScope(newScope).join());
+
+        List<KeyValueTableConfiguration> configList = new ArrayList<>();
+        for (int initialPartition = 5; initialPartition < 8; initialPartition++) {
+            configList.add(KeyValueTableConfiguration.builder().partitionCount(initialPartition).build());
+        }
+
+        List<ResultObserver<CreateKeyValueTableStatus>> observerList = new ArrayList<>(3);
+        for (int i = 0; i < 3; i++) {
+            ResultObserver<CreateKeyValueTableStatus> result = new ResultObserver<>();
+            observerList.add(result);
+        }
+
+        HashMap<Integer, CreateKeyValueTableStatus.Status> resultMap = new HashMap<>(3);
+        List<CompletableFuture<ResultObserver<CreateKeyValueTableStatus>>> requestList = new ArrayList<>();
+        for (int i = 0; i < 3; i++) {
+            final int index = i;
+            CompletableFuture<ResultObserver<CreateKeyValueTableStatus>> requestFutures = CompletableFuture.supplyAsync(() -> {
```

Comment:
There is no guarantee that these requests execute in parallel. For all we know, they could execute serially or in parallel. IMO, this test is not valid.